x/datadeal/keeper/grpc_query_deal.go

```diff
@@ -18,8 +21,55 @@ func (k Keeper) Deal(goCtx context.Context, req *types.QueryDealRequest) (*types
 
 	deal, err := k.GetDeal(ctx, req.DealId)
 	if err != nil {
-		return nil, sdkerrors.ErrKeyNotFound
+		return nil, err
 	}
 
 	return &types.QueryDealResponse{Deal: &deal}, nil
 }
+
+func (k Keeper) DataCert(goCtx context.Context, req *types.QueryDataCertRequest) (*types.QueryDataCertResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	dataCert, err := k.GetDataCert(ctx, req.DealId, []byte(req.DataHash))
```

Comment:
We may have to decode the string using base64 package. 
```suggestion
        bz, err := base64.StdEncoding.DecodeString(req.DataHash)
	if err != nil {
		return nil, err
	}
	dataCert, err := k.GetDataCert(ctx, req.DealId, bz)
```